src/modules/hip/kernel/glitch.hpp

```diff
@@ -0,0 +1,319 @@
+                                      unsigned int *yOffsetB,
+                                      RpptROIPtr roiTensorPtrSrc)
+{
+    int id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if ((id_y >= roiTensorPtrSrc[id_z].xywhROI.roiHeight) || (id_x >= roiTensorPtrSrc[id_z].xywhROI.roiWidth))
+    {
+        return;
+    }
+
+    int xR, yR, xG, yG, xB, yB;
+    xR = id_x + xOffsetR[id_z];
+    yR = id_y + yOffsetR[id_z];
+    xG = id_x + xOffsetG[id_z];
+    yG = id_y + yOffsetG[id_z];
+    xB = id_x + xOffsetB[id_z];
+    yB = id_y + yOffsetB[id_z];
+
+    uint srcIdx = (id_z * srcStridesNH.x) + ((id_y + roiTensorPtrSrc[id_z].xywhROI.xy.y) * srcStridesNH.y) + (id_x + roiTensorPtrSrc[id_z].xywhROI.xy.x) * 3;
+    uint dstIdx = (id_z * dstStridesNCH.x) + (id_y * dstStridesNCH.z) + id_x;
+
+    uint dstIdxR, dstIdxG , dstIdxB;
+    dstIdxR = dstIdx;
+    dstIdxG = dstIdxR + dstStridesNCH.y;
+    dstIdxB = dstIdxG + dstStridesNCH.y;
+
+    uint srcIdxR, srcIdxG, srcIdxB;
+    srcIdxR = srcIdx;
+    srcIdxG = srcIdx + 1;
+    srcIdxB = srcIdx + 2;
+
+    if((yR >= 0) && (yR < roiTensorPtrSrc[id_z].xywhROI.roiHeight) && (xR >= 0) && (xR < roiTensorPtrSrc[id_z].xywhROI.roiWidth))
+        srcIdxR = (id_z * srcStridesNH.x) + ((yR + roiTensorPtrSrc[id_z].xywhROI.xy.y) * srcStridesNH.y) + (xR + roiTensorPtrSrc[id_z].xywhROI.xy.x) * 3;
+
+    if((yG >= 0) && (yG < roiTensorPtrSrc[id_z].xywhROI.roiHeight) && (xG >= 0) && (xG < roiTensorPtrSrc[id_z].xywhROI.roiWidth))
+        srcIdxG = (id_z * srcStridesNH.x) + ((yG + roiTensorPtrSrc[id_z].xywhROI.xy.y) * srcStridesNH.y) + (xG + roiTensorPtrSrc[id_z].xywhROI.xy.x) * 3 + 1;
+
+    if((yB >= 0) && (yB < roiTensorPtrSrc[id_z].xywhROI.roiHeight) && (xB >= 0) && (xB < roiTensorPtrSrc[id_z].xywhROI.roiWidth))
+        srcIdxB = (id_z * srcStridesNH.x) + ((yB + roiTensorPtrSrc[id_z].xywhROI.xy.y) * srcStridesNH.y) + (xB + roiTensorPtrSrc[id_z].xywhROI.xy.x) * 3  + 2;
+
+    dstPtr[dstIdxR] = srcPtr[srcIdxR];
+    dstPtr[dstIdxG] = srcPtr[srcIdxG];
+    dstPtr[dstIdxB] = srcPtr[srcIdxB];
+}
+
+template <typename T>
+__global__ void glitch_pln3_pkd3_tensor(T *srcPtr,
+                                      uint3 srcStridesNCH,
+                                      T *dstPtr,
+                                      uint2 dstStridesNH,
+                                      unsigned int *xOffsetR,
+                                      unsigned int *yOffsetR,
+                                      unsigned int *xOffsetG,
+                                      unsigned int *yOffsetG,
+                                      unsigned int *xOffsetB,
+                                      unsigned int *yOffsetB,
+                                      RpptROIPtr roiTensorPtrSrc)
+{
+
+    int id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
```

Comment:
Why aren't these hip kernels vectorized? Glitch ideally writes elements in sequence, same as the reads.